### PR TITLE
Conditional dispose of HttpClient

### PR DESCRIPTION
**Pull request by alexeyzimarev**
* Dispose `HttpClient` conditionally #1711
* Handling of custom content types #1710
* Clean up values for `Accept` headers #1707

### CODE REVIEW

File: src/RestSharp/RestClient.cs [hunk 1]
```diff
@@ -108,6 +74,57 @@ public partial class RestClient : IDisposable {
...
+        ConfigureHttpClient(HttpClient);
+    }
+
+    /// <summary>
+    /// Creates a new instance of RestSharp using the message handler provided. By default, HttpClient disposes the provided handler
+    /// when the client itself is disposed. If you want to keep the handler not disposed, set disposeHandler argument to false.
+    /// </summary>
+    /// <param name="handler">Message handler instance to use for HttpClient</param>
+    /// <param name="disposeHandler">Dispose the handler when disposing RestClient, true by default</param>
+    public RestClient(HttpMessageHandler handler, bool disposeHandler = true) : this(new HttpClient(handler, disposeHandler)) { }
```
Comment (jnyrup) [inline src/RestSharp/RestClient.cs:98]: I think this should call the other constructor with `, true` to ensure that the constructed `HttpClient` no matter if the injected `handler` should be disposed.

File: src/RestSharp/RestClient.cs [hunk 2]
```diff
@@ -176,5 +193,10 @@ internal void AssignAcceptedContentTypes()
             );
     }
 
-    public void Dispose() => HttpClient.Dispose();
+    public void Dispose() {
```
[the remark this answers is not in the document]
Comment (alexeyzimarev) [reply src/RestSharp/RestClient.cs:196]: I already added `GC.SuppressFinalize(this);`, but I am not sure about the whole thing with `_disposing`. Will check the article.

File: src/RestSharp/RestClient.cs [hunk 3]
```diff
@@ -108,6 +74,57 @@ public partial class RestClient : IDisposable {
     /// <param name="baseUrl"></param>
     public RestClient(string baseUrl) : this(new Uri(Ensure.NotEmptyString(baseUrl, nameof(baseUrl)))) { }
 
+    public RestClient(HttpClient httpClient, RestClientOptions? options = null, bool disposeHttpClient = false) {
+        if (Options?.CookieContainer != null) {
```
[the remark this answers is not in the document]
Comment (alexeyzimarev) [reply src/RestSharp/RestClient.cs:78]: Yes! Thanks.

File: src/RestSharp/RestClient.cs (hunk 1 above)
Comment (alexeyzimarev) [reply src/RestSharp/RestClient.cs:98]: Yeah, you're right. Changed to
```csharp
public RestClient(HttpMessageHandler handler, bool disposeHandler = true) : this(new HttpClient(handler, disposeHandler), null, true) { }
```

File: src/RestSharp/RestClient.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexeyzimarev) [reply src/RestSharp/RestClient.cs:196]: Added a protected virtual `Dispose(bool)`

**Pull request merged**